code/game/turfs/turf.dm

```diff
@@ -265,8 +265,11 @@ var/const/enterloopsanity = 100
 	var/obj/structure/catwalk/catwalk = locate(/obj/structure/catwalk) in src
 	if(catwalk)
 		sound = footstep_sound("catwalk")
-	else
-		sound =  footstep_sound(flooring.footstep_sound)
+	else if(flooring)
```

Comment:
as a precaution - could you make a default value for sound variable, so if something goes wrong it doesn't return null.

Reply:
What happens if someone tries to spacewalk without any floor under him ? Does the default value will default to an actual sound or empty sound ( wasting processing when it can run without sound ) ?